bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/item/ItemResource.java

```diff
@@ -226,6 +226,34 @@ public Response getItems(final @Context UriInfo uriInfo, final @Context HttpHead
         return Response.ok(new Stream2JSONInputStream(itemStream)).build();
     }
 
+    /**
+     *
+     * @param itemname name of the item
+     * @return the namesspace of that item
+     */
+    @GET
+    @RolesAllowed({ Role.USER, Role.ADMIN })
+    @Path("/{itemname: [a-zA-Z_0-9]+}/metadata/namespaces")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Operation(operationId = "getItemNamespaces", summary = "Gets the namespace of an item.", responses = {
+            @ApiResponse(responseCode = "200", description = "OK", content = @Content(schema = @Schema(implementation = String.class))),
+            @ApiResponse(responseCode = "404", description = "Item not found") })
+    public Response getItemNamespaces(@PathParam("itemname") @Parameter(description = "item name") String itemname,
+            @HeaderParam(HttpHeaders.ACCEPT_LANGUAGE) @Parameter(description = "language") @Nullable String language,
+            @QueryParam("exclude") @Parameter(description = "exclude the given namespaces") @Nullable String excludeNamespaces) {
+        Item item = getItem(itemname);
+        final Locale locale = localeService.getLocale(language);
+        final Set<String> excludedNamespaces = splitAndFilterNamespaces(excludeNamespaces, locale);
+
+        if (item != null) {
+            Set<String> allItemMetadataNamespaces = metadataRegistry.getAllNamespaces(itemname).stream()
+                    .filter(n -> !excludedNamespaces.contains(n)).collect(Collectors.toSet());
+            return Response.ok(allItemMetadataNamespaces.toString()).build();
```

Comment:
Maybe use `Collectors.joining(",")` to put them directly in a comma separate string:
```suggestion
            final String allItemMetadataNamespaces = metadataRegistry.getAllNamespaces(itemname).stream()
                    .filter(n -> !excludedNamespaces.contains(n)).collect(Collectors.joining(","));

            return Response.ok(allItemMetadataNamespaces).build();
```

Reply:
I agree with @Hilbrand that you use either `Collectors.joining(",")` or `String.join(",", collection)` and not use `.toString` here. But also see my general comment.